polus-ftl-label-plugin/src/ftl.pyx

```diff
@@ -0,0 +1,966 @@
+        
+    Outputs:
+        numpy.ndarray: Indices of pixel objects
+    """
+
+    ''' Set up AVX vectors '''
+    # Vector to hold pixel values
+    cdef __m128i v
+    # Vector to hold 
+    cdef __m128i mask
+    # Vector for pixel indices
+    cdef __m256i edges
+    # Byte shuffling constants
+    cdef __m128i pshufbcnst = _mm_set_epi8(0x80,  0x80,  0x80,  0x80,
+                                           0x80,  0x80,  0x80,  0x80,
+                                           0x0E,  0x0C,  0x0A,  0x08,
+                                           0x06,  0x04,  0x02,  0x00)
+    cdef __m256i shf        = _mm256_set_epi8(0x80,0x0F,  0x80,0x07,  0x80,0x0E,  0x80,0x06,
+                                              0x80,0x0D,  0x80,0x05,  0x80,0x0C,  0x80,0x04,
+                                              0x80,0x0B,  0x80,0x03,  0x80,0x0A,  0x80,0x02,
+                                              0x80,0x09,  0x80,0x01,  0x80,0x08,  0x80,0x00)
+    # Bit shifts to decode 4-bit packing
+    cdef __m256i shft       = _mm256_set_epi64x(0x04,0x00,0x04,0x00)
+    # 4-bit mask
+    cdef __m256i vmsk       = _mm256_set1_epi8(0x0F)
+    # Indexing
+    cdef __m256i cnst15     = _mm256_set1_epi16(0x000F)
+    cdef __m256i i_vec      = _mm256_set1_epi16(0)
+
+    ''' Constants for bit operations '''
+    # Casts 4-bit boolean to position index
+    cdef unsigned long long cast_index = 0xFEDCBA9876543210
+    # Used to determine if pixel runs were detected
+    cdef unsigned int detect_edge = 0xFFFFFFFE
+    # Clear the lowest value
+    cdef unsigned long long clear_lowest = 0xFFFFFFFFFFFFFFF0
+    # Holds (16) 4-bit indices
+    cdef unsigned long long ind
+
+    ''' Temporary and output vectors '''
+    cdef vector[np.uint16_t] temp
+    temp.resize(16,0)
+    cdef vector[np.uint32_t] output
+    
+    ''' Looping variables '''
+    cdef unsigned long p,i,j,n,r
+    cdef unsigned long position = 0
+    # Whether the last found point was a leading edge of pixels
+    cdef bint on_obj = False
+
+    ''' Dimension and point variables '''
+    cdef unsigned long ndim = len(shape)
+    cdef unsigned long positions
+    cdef unsigned long dim
+    cdef unsigned int [:] coords = np.zeros(ndim-1,dtype=np.uint32)
+    # The number of edges discovered
+    cdef unsigned long count
+
+    ''' Stride information '''
+    cdef unsigned int [:] strides = np.ones(ndim,dtype=np.uint32)
+    for i in range(1,ndim):
+        for j in range(ndim-i):
+            strides[j] = strides[j]*np.uint32(shape[ndim - i])
+    cdef Py_ssize_t last_stride = shape[ndim-1]
+    positions = strides[0]*shape[0]//last_stride
+        
+    '''
+    
+    Loop through all points and find the start and stopping edges of
+    consecutive nonzero values. The way this works is that pixels are laid
+    out linearly in memory according to the last dimension of the matrix.
+    So, in a 2-d matrix pixels in the same row are immediately next to
```

Comment:
I think the official name for this feature is "Data Locality". Good to mention it here.